examples/vercel-ai-chatbot/app/api/chat/route.ts

```diff
@@ -1,34 +1,36 @@
 import { kv } from '@vercel/kv'
 import { StreamingTextResponse } from 'ai'
 import { VellumClient } from 'vellum-ai'
+import { ChatMessage as ChatMessageSerializer } from 'vellum-ai/serialization'
+import { serialization } from 'vellum-ai/core'
 
 import { auth } from '@/auth'
-import { z } from 'zod'
-import { ChatMessageRole } from 'vellum-ai/api/types'
+import {
+  ChatMessage,
+  FunctionCallChatMessageContentValue,
+  WorkflowOutput
+} from 'vellum-ai/api'
+import { nanoid } from 'nanoid'
 
 export const runtime = 'edge'
 
 const vellum = new VellumClient({
-  apiKey: process.env.VELLUM_API_KEY!
+  apiKey: process.env.VELLUM_API_KEY!,
+  environment: {
+    default: 'http://127.0.0.1:8000',
+    predict: 'http://127.0.0.1:8000',
+    documents: 'http://127.0.0.1:8000'
+  }
 })
 
-const zBody = z.object({
-  id: z.string(),
-  messages: z
-    .object({
-      role: z
-        .enum(['user', 'assistant', 'function', 'system', 'tool'])
-        .transform<ChatMessageRole>(s =>
-          s === 'tool' ? 'FUNCTION' : (s.toUpperCase() as ChatMessageRole)
-        ),
-      content: z.string()
-    })
-    .array()
+const requestBodySerializer = serialization.object({
+  id: serialization.string(),
+  messages: serialization.list(ChatMessageSerializer)
 })
```

Comment:
This is cool. Users could use our fern-generated api sdks to import serializers that could serialize their _own_ requests!